ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/queries/imageDetails.ts

```diff
@@ -0,0 +1,50 @@
+import { gql } from '@apollo/client';
+
+export type ImageDetailsVariables = {
+    id: string;
+};
+
+export type ImageDetailsResponse = {
+    image: {
+        deploymentCount: number;
+        name: {
+            fullName: string;
+        } | null;
+        operatingSystem: string;
+        metadata: {
+            v1: {
+                created: Date | null;
+                digest: string;
+            } | null;
+        } | null;
+
+        scan: {
+            dataSource: { name: string };
+            scanTime: Date | null;
+        };
+    };
+};
+
+export const imageDetailsQuery = gql`
+    query getImageDetails($id: ID!) {
+        image(id: $id) {
+            deploymentCount
+            name {
+                fullName
+            }
+            operatingSystem
+            metadata {
+                v1 {
+                    created
+                    digest
+                }
+            }
+            scan {
+                dataSource {
+                    name
+                }
+                scanTime
+            }
+        }
+    }
+`;
```

Comment:
it might be useful to have the queries closer to the components using the queries if we don't have a need to recycle them in other components. if we do need to reuse them, we can probably create a `queries` directory to hold them for easy discovery? 

either way, it feels odd to me to mix the type and the gql strings here :/

Reply:
Yeah I still feel like finding a place for the queries and types for GraphQL is a little awkward, compared to our REST services which is pretty clear cut. I'll move the type and query right into the component for now and we can worry about common fragments later.

I'm also going to take a look at GraphQL type generation soon since it seems like we shouldn't have to manually create types for these queries everywhere.